src/app.md

```diff
@@ -271,6 +272,54 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-file-skip-dot] #ex-file-skip-dot
+<a name="ex-file-skip-dot"></a>
+##  Traverse directories while skipping dotfiles
+
+Iterate over the entries that are not hidden files or hidden directories. 
+
+[`WalkDir::filter_entry`] descend recursively entries passing the predicate whereas [`std::iter::Iterator::filter`] would by applied to each `DirEntry` even if the parent is a hidden directory. [`WalkDir::depth`] is used to skip the level 0 which is the current directory `"."`.
+
+[![walkdir-badge]][walkdir] [![cat-filesystem-badge]][cat-filesystem]
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate walkdir;
+
+use walkdir::{DirEntry, WalkDir, WalkDirIterator};
+#
+# error_chain! {
+#     foreign_links {
+#         WalkDir(walkdir::Error);
+#         Io(std::io::Error);
+#     }
+# }
+#
+
+fn is_not_hidded(entry: &DirEntry) -> bool {
+    !entry
+         .file_name()
+         .to_str()
+         .map(|s| entry.depth() > 0 && s.starts_with("."))
+         .unwrap_or(false)
+}
+
+fn run() -> Result<()> {
+    for result in WalkDir::new(".")
+            .into_iter()
+            .filter_entry(|e| is_not_hidded(e))
+            .filter_map(|v| v.ok()) {
+        println!("{}", result.path().display());
+        println!("{}", result.file_name().to_str().map(|s| s).unwrap_or(""));
```

Comment:
there would be few ways depending what you want to achieve. If you would like to obtain the `file_name`s collected into a vec you would use further iterator transforms but if you are just for the sideffects (like printing) `match` or `if let` for the Ok value would suffice.